# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/scene_building.rs
# Commit: aaed31bbe8f4
# Full Hash: aaed31bbe8f4e0a8892e632633c27033549eaf33
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2020-03-19 03:53:34
# Regressor Bug: 1614655
# File Overlap Count: 2
# Description:
#   Bug 1614655 - Part 2: Allow 1:n mapping of Gecko - WR display items r=jrmuizel
#   
#   This patch changes the underlying storage for WR display items in DisplayItemCache
#   from Vec<Option<CachedDisplayItem> to Vec<Vec<CachedDisplayItem>>.
#   This allows storing multiple WebRender display items for one Gecko display item.
# ==============================================================================

diff -r 0e791acac783 -r aaed31bbe8f4 gfx/wr/webrender/src/scene_building.rs
--- a/gfx/wr/webrender/src/scene_building.rs	Wed Mar 18 23:46:27 2020 +0000
+++ b/gfx/wr/webrender/src/scene_building.rs	Wed Mar 18 23:47:05 2020 +0000
@@ -1510,8 +1510,9 @@
                 unreachable!("Should have returned in parent method.")
             }
 
-            DisplayItem::ReuseItem(..) => {
-                unreachable!("Iterator logic error")
+            DisplayItem::ReuseItems(key) |
+            DisplayItem::RetainedItems(key) => {
+                unreachable!("Iterator logic error: {:?}", key);
             }
 
             DisplayItem::PushShadow(info) => {